app/Libraries/Opengraph/CommentOpengraph.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+declare(strict_types=1);
+
+namespace App\Libraries\Opengraph;
+
+use App\Models\Comment;
+
+class CommentOpengraph
+{
+    public function __construct(private Comment $comment)
+    {
+    }
+
+    public function get(): array
+    {
+        $user = $this->comment->user;
+
+        return priv_check('CommentShow', $this->comment)->can()
```

Comment:
priv_check_user with null user?

Reply:
originally, I had everything assume opengrapgh would be viewed only by `null` user, but if you go to the page with a deleted comment as a moderator, you see a different thing, so I'm not so sure now 👀